src/url.rs

```diff
@@ -0,0 +1,387 @@
+                            match dump_graph_as_turtle(&metrics_store) {
+                                Ok(bytes) => {
+                                    // Create MQA event
+                                    match std::str::from_utf8(bytes.as_slice()) {
+                                        Ok(turtle) => Ok(Some(MQAEvent {
+                                            event_type: MQAEventType::UrlsChecked,
+                                            fdk_id: fdk_id.clone(),
+                                            graph: turtle.to_string(),
+                                            timestamp: Utc::now().timestamp_millis(),
+                                        })),
+                                        Err(e) => Err(format!(
+                                            "{} - Failed dumping graph as turtle: {}",
+                                            fdk_id, e
+                                        )),
+                                    }
+                                }
+                                Err(e) => Err(format!(
+                                    "{} - Failed dumping graph as turtle: {}",
+                                    fdk_id, e
+                                )),
+                            }
+                        }
+                        Err(e) => Err(format!(
+                            "{} - Failed creating new metrics store: {}",
+                            fdk_id, e
+                        )),
+                    }
+                }
+                None => Err(format!("{} - Dataset node not found in graph", fdk_id)),
+            }
+        }
+        Err(e) => Err(format!("{} - Failed to parse graph: {}", fdk_id, e)),
+    }
+}
+
+fn check_urls(fdk_id: String, dataset_node: NamedNodeRef, store: &Store) -> Result<Store, Error> {
+    // Make MQA metrics model (DQV)
+    let metrics_store = create_metrics_store(dataset_node)?;
+    let distributions =
+        list_distributions(dataset_node, store).collect::<Result<Vec<Quad>, _>>()?;
+
+    for dist in distributions {
+        metrics_store.insert(
+            Quad::new(
+                dist.subject.clone(),
+                dist.predicate.clone(),
+                dist.object.clone(),
+                GraphName::DefaultGraph,
+            )
+            .as_ref(),
+        )?;
+
+        let dist_node_option = match dist.object {
+            Term::NamedNode(n) => Some(NamedOrBlankNode::NamedNode(n)),
+            Term::BlankNode(n) => Some(NamedOrBlankNode::BlankNode(n)),
+            _ => None,
+        };
+
+        match dist_node_option {
+            Some(dist_node) => {
+                info!("{} - Extracting urls from distribution", fdk_id);
+                match extract_urls_from_distribution(dist_node.as_ref(), &store) {
+                    Ok(urls) => {
+                        info!("{} - Number of urls found {}", fdk_id, urls.len());
+
+                        for url in urls {
```

Comment:
Når du ikke gjør noe med `Err` casen er det like greit å bare bruke `?`.
```suggestion
                let urls = extract_urls_from_distribution(dist_node.as_ref(), &store)?;
                info!("{} - Number of urls found {}", fdk_id, urls.len());

                for url in urls {
```